### PR TITLE
PR: Limit the number of lines in history according to the relevant setting

### LINKED ISSUE #7080: Spyder is slow to start due to big history file
When starting, spyder reads a 10Mo text file in ~/.spyder-py3 called history.py.
That takes ~15s, which is a painfully noticeable start-up time. 
Is there an option to limit how far back I want to load the history?

The easy work-around is for me to edit manually this file, but if an user is not aware of this, he might get annoyed by the start-up time.
I can go from 22s to 6s start-up time, which is not fast but way less annoying.

Also closing spyder takes almost 9s. I couldn't figure out why. If I have to restart spyder, usually I just kill it to avoid having to wait.

Python 3.6.5 64bits, Qt 5.10.1, PyQt5 5.10.1 on Darwin 
OSX 10.13.4

### PR DESCRIPTION
**Pull request by impact27**
### Pull Request Checklist

* [x] Read and followed this repo's [Contributing Guidelines](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md)
* [x] Based your PR on the latest version of the correct branch (master or 3.x)
* [x] Followed [PEP8](https://www.python.org/dev/peps/pep-0008/) for code style
* [x] Ensured your pull request hasn't eliminated unrelated blank lines/spaces,
      modified the ``spyder/defaults`` directory, or added new icons/assets
* [x] Wrote at least one-line docstrings for any new functions
* [x] Added at least one unit test covering the changes, if at all possible
* [x] Described your changes and the motivation for them below
* [x] Noted what issue(s) this pull request resolves, creating one if needed
* [x] ~~Included a screenshot, if this PR makes any visible changes to the UI~~

## Description of Changes

A better thing would be to read the history from ipython, but I am not sure that this would work in every cases.

### Issue(s) Resolved

Fixes #7080

### CODE REVIEW

**Comment by CAM-Gerlach** [comment]
Thanks for doing this! I need to head to work now, but a few quick things:

* Please briefly describe what your PR actually does in the title and the description, so we can follow it easily
* Fix the PEP8 issue
* The test failures are legitimate, and it looks like its due to a typo in your code (see comment). It's easy to make, but if you're using Spyder or had actually tested the change, it would have been easy to catch
* Can you add a test for this?
* Not sure why you deleted the PR template that specified several of these things in a checklist...

File: spyder/plugins/history.py [hunk 1]
```diff
@@ -230,6 +232,11 @@ def add_history(self, filename):
         editor.toggle_wrap_mode( self.get_option('wrap') )
 
         text, _ = encoding.read(filename)
+        linebreaks = [m.start() for m in re.finditer('\n', text)]
+        maxNline = CONF.get('historylog', 'max_entries')
+        if len(linebreak) > maxNline:
```
Comment (CAM-Gerlach) [inline spyder/plugins/history.py]: This is where all the tests are failing--it should be ``linebreaks`` presumably. Easy mistake to make, but one a basic code checker or a simple test (unit test or just actually testing that the functionality worked without error) would have caught.

File: spyder/plugins/history.py [hunk 2]
```diff
@@ -29,6 +30,7 @@
 from spyder.widgets.tabs import Tabs
 from spyder.widgets.sourcecode import codeeditor
 from spyder.widgets.findreplace import FindReplace
+from spyder.config.main import CONF
```
Comment (ccordoba12) [inline spyder/plugins/history.py]: Please remove this. It's not needed.

File: spyder/plugins/history.py [hunk 3]
```diff
@@ -230,6 +232,11 @@ def add_history(self, filename):
         editor.toggle_wrap_mode( self.get_option('wrap') )
 
         text, _ = encoding.read(filename)
+        linebreaks = [m.start() for m in re.finditer('\n', text)]
+        maxNline = CONF.get('historylog', 'max_entries')
```
Comment (ccordoba12) [inline spyder/plugins/history.py]: Plugins can access config their own options like this

    maxNline = self.get_option('max_entries')

**Comment by ccordoba12** [comment]
Things look good, I just left a minor comment.

**Comment by CAM-Gerlach** [comment]
@ccordoba12 Do we want a regression test for this to ensure the change is and remains effective?

**Comment by ccordoba12** [comment]
A test would be nice, yes, but it doesn't seem easy to implement. Besides, the code looks good.

File: spyder/plugins/history.py [hunk 4]
```diff
@@ -230,6 +231,11 @@ def add_history(self, filename):
         editor.toggle_wrap_mode( self.get_option('wrap') )
 
         text, _ = encoding.read(filename)
+        linebreaks = [m.start() for m in re.finditer('\n', text)]
```
Comment (ccordoba12) [inline spyder/plugins/history.py]: We need to convert all EOL's to `\n` before doing this.

Comment (CAM-Gerlach) [reply spyder/plugins/history.py]: Can't we just always use ``\n`` for our own history files, independent of the platform?

**Pull request merged**